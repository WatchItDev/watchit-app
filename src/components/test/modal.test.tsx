import { MemoryRouter } from 'react-router';

// Testing libraries
import { describe, it, expect, afterEach } from 'vitest';
import { cleanup, render } from '@testing-library/react';

// Base testing component
import Modal, {ModalProps} from "@src/components/modal.tsx";

const RenderComponent = (props: ModalProps) => render(<MemoryRouter><Modal {...props} /></MemoryRouter>);

describe('Modal component testing', () => {
  const modalProps = {
    onClose: () => {},
    title: 'Modal title',
    open: true,
    renderContent: <div>Modal content</div>
  }

  afterEach(() => {
    cleanup();
  })

  it('to match snapshot', () => {
    const { asFragment } = RenderComponent(modalProps);
    expect(asFragment()).toMatchSnapshot();
  });

  // Test if title is equal to 'Modal title'
  it('title is equal to "Modal title"', () => {
    const { getByText } = RenderComponent(modalProps);
    expect(getByText('Modal title')).toBeInTheDocument();
  });

  // Pass false to open prop and check if not "Modal title" is in the document
  it('title is not in the document', () => {
    const { queryByText } = RenderComponent({ ...modalProps, open: false });
    expect(queryByText('Modal title')).not.toBeInTheDocument();
  });
});
