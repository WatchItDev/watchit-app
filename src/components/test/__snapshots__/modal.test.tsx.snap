// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Modal component testing > to match snapshot 1`] = `<DocumentFragment />`;
